consensus/ballot_state.go

```diff
@@ -529,21 +531,59 @@ func (s *BallotState) RelevantRange(x SlotValue) (int, int) {
 	min, max := 0, 0
 	for _, message := range s.M {
 		a, b := message.RelevantRange(x)
-		min, max = RangeUnion(min, max, a, b) 
+		min, max = RangeUnion(min, max, a, b)
 	}
 	return min, max
 }
 
+// Returns the max ballot number that a blocking set of nodes are talking about.
+func (s *BallotState) MaxActionableBallotNumber() int {
+	numberToNodes := make(map[int][]string)
+
+	for node, message := range s.M {
+		maxN := message.MaxN()
+		numberToNodes[maxN] = append(numberToNodes[maxN], node)
+	}
+
+	i := 0
```

Comment:
this loop confused me at first, can you avoid the use of i by doing like ns := []int{} and then ns = append(ns, n) inside the loop? i think that is more go-styley

Reply:
I think my approach is more efficient because the size is known a priori.

Reply:
pfff, efficiency when creating an array of size 4. fine